### ISSUE TITLE
Cruise-Control-Metrics-Reporter should not be blocked if JVM's CPU usage is unavailable

### ISSUE THREAD

**Issue by Sidong-Wei** [linkedin/cruise-control#1670]
This part of [code](https://github.com/linkedin/cruise-control/blob/master/cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/metric/MetricsUtils.java#L400) will throw an exception if the `getProcessCpuLoad()` call returns a negative value and the current JVM CPU usage is unavailable, and would thus block the MetricsReporter from sending metrics. This is a part of the error message that I got:
```
[2021-08-26 15:09:40,207] ERROR Got exception in Cruise Control metrics reporter (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
java.lang.IllegalStateException: Java Virtual Machine recent CPU usage is not available.
	at com.linkedin.kafka.cruisecontrol.metricsreporter.metric.MetricsUtils.getCpuMetric(MetricsUtils.java:400)
	at com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter.reportCpuUtils(CruiseControlMetricsReporter.java:414)
	at com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter.run(CruiseControlMetricsReporter.java:333)
	at java.base/java.lang.Thread.run(Thread.java:866)
[2021-08-26 15:09:41,292] WARN [Producer clientId=CruiseControlMetricsReporter] Got error produce response with correlation id 12 on topic-partition __CruiseControlMetrics-0, retrying (4 attempts left). Error: NOT_LEADER_OR_FOLLOWER (org.apache.kafka.clients.producer.internals.Sender)
[2021-08-26 15:09:41,292] WARN [Producer clientId=CruiseControlMetricsReporter] Received invalid metadata error in produce request on partition __CruiseControlMetrics-0 due to org.apache.kafka.common.errors.NotLeaderOrFollowerException: For requests intended only for the leader, this error indicates that the broker is not the current leader. For requests intended for any replica, this error indicates that the broker is not a replica of the topic partition.. Going to request metadata update now (org.apache.kafka.clients.producer.internals.Sender)
[2021-08-26 15:09:41,433] WARN [Producer clientId=CruiseControlMetricsReporter] Got error produce response with correlation id 14 on topic-partition __CruiseControlMetrics-0, retrying (3 attempts left). Error: NOT_LEADER_OR_FOLLOWER (org.apache.kafka.clients.producer.internals.Sender)
[2021-08-26 15:09:41,433] WARN [Producer clientId=CruiseControlMetricsReporter] Received invalid metadata error in produce request on partition __CruiseControlMetrics-0 due to org.apache.kafka.common.errors.NotLeaderOrFollowerException: For requests intended only for the leader, this error indicates that the broker is not the current leader. For requests intended for any replica, this error indicates that the broker is not a replica of the topic partition.. Going to request metadata update now (org.apache.kafka.clients.producer.internals.Sender)
[2021-08-26 15:09:41,548] WARN [Producer clientId=CruiseControlMetricsReporter] Got error produce response with correlation id 16 on topic-partition __CruiseControlMetrics-0, retrying (2 attempts left). Error: NOT_LEADER_OR_FOLLOWER (org.apache.kafka.clients.producer.internals.Sender)
[2021-08-26 15:09:41,548] WARN [Producer clientId=CruiseControlMetricsReporter] Received invalid metadata error in produce request on partition __CruiseControlMetrics-0 due to org.apache.kafka.common.errors.NotLeaderOrFollowerException: For requests intended only for the leader, this error indicates that the broker is not the current leader. For requests intended for any replica, this error indicates that the broker is not a replica of the topic partition.. Going to request metadata update now (org.apache.kafka.clients.producer.internals.Sender)
[2021-08-26 15:09:41,707] WARN [Producer clientId=CruiseControlMetricsReporter] Got error produce response with correlation id 18 on topic-partition __CruiseControlMetrics-0, retrying (1 attempts left). Error: NOT_LEADER_OR_FOLLOWER (org.apache.kafka.clients.producer.internals.Sender)
[2021-08-26 15:09:41,707] WARN [Producer clientId=CruiseControlMetricsReporter] Received invalid metadata error in produce request on partition __CruiseControlMetrics-0 due to org.apache.kafka.common.errors.NotLeaderOrFollowerException: For requests intended only for the leader, this error indicates that the broker is not the current leader. For requests intended for any replica, this error indicates that the broker is not a replica of the topic partition.. Going to request metadata update now (org.apache.kafka.clients.producer.internals.Sender)
[2021-08-26 15:09:41,818] WARN [Producer clientId=CruiseControlMetricsReporter] Got error produce response with correlation id 20 on topic-partition __CruiseControlMetrics-0, retrying (0 attempts left). Error: NOT_LEADER_OR_FOLLOWER (org.apache.kafka.clients.producer.internals.Sender)
[2021-08-26 15:09:41,818] WARN [Producer clientId=CruiseControlMetricsReporter] Received invalid metadata error in produce request on partition __CruiseControlMetrics-0 due to org.apache.kafka.common.errors.NotLeaderOrFollowerException: For requests intended only for the leader, this error indicates that the broker is not the current leader. For requests intended for any replica, this error indicates that the broker is not a replica of the topic partition.. Going to request metadata update now (org.apache.kafka.clients.producer.internals.Sender)
[2021-08-26 15:09:41,988] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_REQUEST_HANDLER_AVG_IDLE_PERCENT,time=1629990578164,brokerId=0,value=0.984] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,988] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_RESPONSE_QUEUE_SIZE,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,988] WARN Failed to send Cruise Control metric [BROKER_METRIC,ALL_TOPIC_REPLICATION_BYTES_OUT,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,988] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_PRODUCE_REQUEST_QUEUE_TIME_MS_MAX,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,989] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_PRODUCE_REQUEST_QUEUE_TIME_MS_MEAN,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,989] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_PRODUCE_REQUEST_QUEUE_TIME_MS_50TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,989] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_PRODUCE_REQUEST_QUEUE_TIME_MS_999TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,989] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_FOLLOWER_FETCH_REQUEST_QUEUE_TIME_MS_MAX,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,989] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_FOLLOWER_FETCH_REQUEST_QUEUE_TIME_MS_MEAN,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,989] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_FOLLOWER_FETCH_REQUEST_QUEUE_TIME_MS_50TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,989] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_FOLLOWER_FETCH_REQUEST_QUEUE_TIME_MS_999TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,989] WARN Failed to send Cruise Control metric [BROKER_METRIC,ALL_TOPIC_BYTES_OUT,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,989] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_PRODUCE_LOCAL_TIME_MS_MAX,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,989] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_PRODUCE_LOCAL_TIME_MS_MEAN,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,990] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_PRODUCE_LOCAL_TIME_MS_50TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,990] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_PRODUCE_LOCAL_TIME_MS_999TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,990] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_CONSUMER_FETCH_REQUEST_QUEUE_TIME_MS_MAX,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,994] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_CONSUMER_FETCH_REQUEST_QUEUE_TIME_MS_MEAN,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,994] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_CONSUMER_FETCH_REQUEST_QUEUE_TIME_MS_50TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,995] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_CONSUMER_FETCH_REQUEST_QUEUE_TIME_MS_999TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,995] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_RESPONSE_QUEUE_SIZE,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,995] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_RESPONSE_QUEUE_SIZE,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,995] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_RESPONSE_QUEUE_SIZE,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,995] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_CONSUMER_FETCH_TOTAL_TIME_MS_MAX,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,995] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_CONSUMER_FETCH_TOTAL_TIME_MS_MEAN,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,995] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_CONSUMER_FETCH_TOTAL_TIME_MS_50TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,995] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_CONSUMER_FETCH_TOTAL_TIME_MS_999TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,995] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_REQUEST_QUEUE_SIZE,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,995] WARN Failed to send Cruise Control metric [BROKER_METRIC,ALL_TOPIC_REPLICATION_BYTES_IN,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,995] WARN Failed to send Cruise Control metric [BROKER_METRIC,ALL_TOPIC_PRODUCE_REQUEST_RATE,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,995] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_FOLLOWER_FETCH_LOCAL_TIME_MS_MAX,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,996] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_FOLLOWER_FETCH_LOCAL_TIME_MS_MEAN,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,996] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_FOLLOWER_FETCH_LOCAL_TIME_MS_50TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,996] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_FOLLOWER_FETCH_LOCAL_TIME_MS_999TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,996] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_CONSUMER_FETCH_LOCAL_TIME_MS_MAX,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,996] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_CONSUMER_FETCH_LOCAL_TIME_MS_MEAN,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,996] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_CONSUMER_FETCH_LOCAL_TIME_MS_50TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,996] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_CONSUMER_FETCH_LOCAL_TIME_MS_999TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,996] WARN Failed to send Cruise Control metric [BROKER_METRIC,ALL_TOPIC_MESSAGES_IN_PER_SEC,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,996] WARN Failed to send Cruise Control metric [BROKER_METRIC,ALL_TOPIC_FETCH_REQUEST_RATE,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,996] WARN Failed to send Cruise Control metric [BROKER_METRIC,ALL_TOPIC_BYTES_IN,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,996] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_PRODUCE_TOTAL_TIME_MS_MAX,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,996] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_PRODUCE_TOTAL_TIME_MS_MEAN,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,996] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_PRODUCE_TOTAL_TIME_MS_50TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,996] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_PRODUCE_TOTAL_TIME_MS_999TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,996] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_FOLLOWER_FETCH_TOTAL_TIME_MS_MAX,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,997] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_FOLLOWER_FETCH_TOTAL_TIME_MS_MEAN,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,997] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_FOLLOWER_FETCH_TOTAL_TIME_MS_50TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,997] WARN Failed to send Cruise Control metric [BROKER_METRIC,BROKER_FOLLOWER_FETCH_TOTAL_TIME_MS_999TH,time=1629990578164,brokerId=0,value=0.000] (com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter)
[2021-08-26 15:09:41,997] WARN [Producer clientId=CruiseControlMetricsReporter] Received invalid metadata error in produce request on partition __CruiseControlMetrics-0 due to org.apache.kafka.common.errors.NotLeaderOrFollowerException: For requests intended only for the leader, this error indicates that the broker is not the current leader. For requests intended for any replica, this error indicates that the broker is not a replica of the topic partition.. Going to request metadata update now (org.apache.kafka.clients.producer.internals.Sender)
```

However, according to Java [document](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.management/com/sun/management/OperatingSystemMXBean.html#getProcessCpuLoad()), a negative return value of `getProcessCpuLoad()` method only means `recent CPU usage is not available` without any other implication, so it should not be a critical issue and the server should not be blocked by this issue.

I think it will be better if code is changed here so no exception will be raised when a negative value is returned, but rather just log it for reference. The reason I am raising this issue is that when I tested CruiseControl using AdoptOpenJDK11 with Openj9, `getProcessCpuLoad()` will return `-1` on some occasions (and the server gets blocked). I also believe there could be other scenarios where the same issue happens on other JRE, OS, or hardware, but by itself, it is not implying any real critical issues, so the Kafka servers and MetricsReporters should continue to work as normal.

**Comment by Sidong-Wei** [comment]
@efeg Thanks for raising the PR, and thanks for the explanation. Yeah, I realized later that for OpenJ9 JVM there is another issue that is actually blocking the metrics reporter, and these two issues happen at the same time (first-time topic generation) so I thought it was one issue. I will check on the other issue, thanks again
